core/src/network/handler/lists.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{
+    action::{Action, ActionWrapper},
+    context::Context,
+    entry::CanPublish,
+    instance::dispatch_action,
+    network::handler::{get_content_aspect, get_meta_aspects},
+};
+use holochain_core_types::error::HcResult;
+use holochain_net::connection::json_protocol::{EntryListData, GetListData};
+use holochain_persistence_api::cas::content::{Address, AddressableContent};
+use snowflake::ProcessUniqueId;
+use std::{collections::HashMap, sync::Arc, thread};
+
+pub fn handle_get_authoring_list(get_list_data: GetListData, context: Arc<Context>) {
+    thread::Builder::new()
+        .name(format!(
+            "handle_authoring_list/{}",
+            ProcessUniqueId::new().to_string()
+        ))
+        .spawn(move || {
+            let mut address_map = HashMap::new();
+            for entry in get_all_public_chain_entries(context.clone()) {
+                address_map.insert(
+                    entry.clone(),
+                    get_all_aspect_addresses(&entry, context.clone())
+                        .expect("Error getting entry aspects of authoring list"),
```

Comment:
Can we really have an expect here?  It seems possible that this could fail.

Reply:
Yep, and then the thread that we spawned 6 lines above should stop. This expect won't make the main thread panic.